### ISSUE TITLE
java.lang.RuntimeException: Interaction with plugin with id 'github.pr' implementing 'scm' extension failed while requesting for 'latest-revisions-since'. Reason: [Unsuccessful response from plugin. Plugin returned with code '500' and the following response: '"Exception caught during execution of reset command. {0}"']

### ISSUE THREAD

**Issue by rzenker** [ashwanthkumar/gocd-build-github-pull-requests#8]
I'm not sure where this exception is being raised from in the code.  Any hints you can offer to help debug this?

**Comment by ashwanthkumar** [comment]
Can you look for the stack trace in the log file? We're looking for a file `plugin-github.pr.log` in the agent's home directory.

**Comment by rzenker** [comment]
I believe I've tracked this down to the flyweight dir getting into a horribly inconsistent state.  I've noticed that it is attempting to reset it to various merges that are still in github, but some are really old/closed/never merged pull requests (which can be really far from current head and likely not automatically mergable).   Is it possible to filter this on open PRs only and skip this test for closed PRs for now?  (or trap for mergability)

**Comment by rzenker** [comment]
Didn't look like you needed the checkout here: https://github.com/ashwanthkumar/gocd-build-github-pull-requests/blob/master/src/main/java/in/ashwanthkumar/gocd/github/GitHubPRBuildPlugin.java#L204

If not, it may be just removing that line will keep the working checkout clean without all the weirdness with checking out vastly different commits.

**Comment by rzenker** [comment]
https://github.com/ashwanthkumar/gocd-build-github-pull-requests/pull/12 is working so far for me.

**Comment by ashwanthkumar** [comment]
Handling only open PR is on my list, will fix that.

Checkout there is used for computing the new commits that happened for every PR since the last build. When we do [cloneOrFetch](https://github.com/ashwanthkumar/gocd-build-github-pull-requests/blob/master/src/main/java/in/ashwanthkumar/gocd/github/JGitHelper.java#L37) we clean the working directory every time. 

Unless Go is sending the last-modified-since in parallel, I don't see how the flyweight directory could go into an inconsistent state. Are you able to replicate the problem consistently across all your repositories? If yes, can you help me to replicate the problem?

**Comment by rzenker** [comment]
Haven't had a chance to reproduce since I got it working for this one repo.  It had a long history of PRs (some that never got merged and were closed).  What I saw happening in the flyweight dir was the working dir getting into inconsistent state with a bunch of untracked/old directory/file layouts so that's what led me to use the clean before each checkout and I haven't seen it since.  Had a case where a dir became a submodule in the past, but an old closed PR had a bad link that the new checkout couldn't overwrite.   

Any concerns with the extra clean before checkout?  Seemed elegant to me.. Just had to make that method public.
